lib/channels/src/index.ts

```diff
@@ -0,0 +1,152 @@
+export type ChannelHandler<TEventArgs = any> = (event: ChannelEvent<TEventArgs>) => void;
+
+export interface ChannelTransport<TEventArgs = any> {
+  send(event: ChannelEvent<TEventArgs>): void;
+  setHandler(handler: ChannelHandler<TEventArgs>): void;
+}
+
+export interface ChannelEvent<TEventArgs = any> {
+  type: string; // eventName
```

Comment:
Are the possible values for this `type` field of a limited set? If so it might be better to turn this into a Enum. This depends on whether we have full knowledge on the possible values for this field and also if that's something of importance of course.

Reply:
Would be nice, but I don't think it's possible, since addons are able to use the channel too.